cfme/tests/candu/test_gap_collection.py

```diff
@@ -0,0 +1,88 @@
+import pytest
+from datetime import datetime, timedelta
+
+from cfme import test_requirements
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.wait import wait_for
+
+
+pytestmark = [
+    pytest.mark.tier(3),
+    test_requirements.c_and_u,
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.provider([VMwareProvider],
+    scope='module',
+    required_fields=[(['cap_and_util', 'capandu_vm'], 'cu-24x7')])
+]
+
+
+ELEMENTS = ['vm', 'host']
+GRAPH_TYPE = ['hourly', 'daily']
+
+
+@pytest.fixture(scope='module')
+def order_data(appliance, provider, enable_candu):
+    # Order two day back gap collection data for testing
+    end_date = datetime.now()
+    start_date = end_date - timedelta(days=2)
+
+    view = navigate_to(appliance.server.zone, 'CANDUGapCollection')
+    view.candugapcollection.end_date.fill(end_date)
+    view.candugapcollection.start_date.fill(start_date)
+    view.candugapcollection.submit.click()
+
+
+@pytest.mark.parametrize('graph_type', GRAPH_TYPE)
+@pytest.mark.parametrize('element', ELEMENTS)
+def test_gap_collection(appliance, provider, element, graph_type, order_data):
+    """ Test gap collection data
+
+    prerequisites:
+        * C&U enabled appliance
+
+    Steps:
+        * Navigate to Configuration > Diagnostics > Zone Gap Collection Page
+        * Order old data
+        * Navigate to VM or Host Utilization page
+        * Check for Hourly data
+        * Check for Daily data
+    """
+    if element == 'host':
```

Comment:
Required: This looks like a setup, Suggest moving it to parametrize fixture which will return you a correct object
line 51-59

Reply:
As per our discussion on IRC, fixture parameter will run before provider. So, better to go with same. Its faster (for one provider run all `vm-hourly`, `vm-daily`, and `host-hourly`, `host-daily`) and nice id.